operators/vision/image_encoder_darwin.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <CoreServices/CoreServices.h>
+#include <ImageIO/ImageIO.h>
+
+#include "op_def_struct.h"
+#include "ext_status.h"
+
+namespace ort_extensions::internal {
+
+struct EncodeImage {
+  OrtxStatus OnInit() {
+    float compression = 0.95;
+
+    CFStringRef optionKeys[2];
+    CFTypeRef optionValues[2];
+    optionKeys[0] = kCGImagePropertyHasAlpha;
+    optionValues[0] = (CFTypeRef)kCFBooleanFalse;
+    optionKeys[1] = kCGImageDestinationLossyCompressionQuality;
```

Comment:
`FilterOption` for PNG is not exposed on Darwin. Only `LossyCompressionQuality` is available.